pkg/template_generator.go

```diff
@@ -130,6 +130,19 @@ func generateTemplateForPodSpec(podSpec kapi.PodSpec, key string, value map[stri
 		podSpec.RestartPolicy = kapi.RestartPolicy(fmt.Sprintf("{{.Values.%s.%s}}", key, RestartPolicy))
 	}
 
+	if len(podSpec.ImagePullSecrets) != 0 {
+		imagePullSecretsObj := []kapi.LocalObjectReference{}
+
+		for _, imagePullSecrets := range podSpec.ImagePullSecrets {
+			imagePullSecrets.Name = generateSafeKey(imagePullSecrets.Name)
+			value["imagePullSecrets"] = imagePullSecrets.Name
```

Comment:
Yes, this is what i did. Only thing is the name of secret.
One question: Any reason to change resources metadata name with generateSafeKey() & '{{ template "fullname" . }}-name from original input yaml?
This just creates confusion on what i had in original yaml & what resource name will be in Charts.

For ex. in Secrets, if i create it in different Chart created by Chartify & reference the original name in deployment chart, it will not work. First i need to get new name generated in Secret Charts & then update deployment Charts values.yml.
Also, for Service names, i have few automation that get the details of service using java k8s apis. but as the name changes everytime, it won't work.

If we keep naming as it was in original input yaml, it will be less confusing with no complex logic to sync between different Charts.